### PR TITLE
Chronograf alert endpoint updates

### LINKED ISSUE #2722: New alert endpoints in Chronograf
**PRs:**

- https://github.com/influxdata/chronograf/pull/5675
- https://github.com/influxdata/chronograf/pull/5682
- https://github.com/influxdata/chronograf/pull/5740
- https://github.com/influxdata/chronograf/pull/5752

Add the following alert notification endpoints:

- [ ] ServiceNow
- [ ] BigPanda
- [ ] Microsoft Teams
- [ ] Zenoss

### LINKED ISSUE #2725: Remove HipChat support from Chronograf
**PR:** https://github.com/influxdata/chronograf/pull/5700

### PR DESCRIPTION
**Pull request by sanderson**
Closes #2722 
Closes #2725 

- [x] Tests pass (no build errors)
- [x] Rebased/mergeable

### CODE REVIEW

File: content/chronograf/v1.9/guides/configuring-alert-endpoints.md [hunk 1]
```diff
@@ -60,101 +58,91 @@ For example, Chronograf's Slack integration allows users to specify a default ch
...
+1. In the **Configure Alert Endpoints** of the **Configure Kapacitor Connection** page,
+   click the **BigPanda** tab.
+2. Enter the following:
 
-![HipChat token](/img/chronograf/1-6-g-eventhandlers-hipchattoken.png)
+    - **URL**: BigPanda [alerts API URL](https://docs.bigpanda.io/reference#alerts-how-it-works).
+      Default is `https://api.bigpanda.io/data/v2/alerts`.
+    - **Token**: BigPanda [API Authorization token (API key)](https://docs.bigpanda.io/docs/api-key-management).
+    - **Application Key**: BigPanda [App Key](https://docs.bigpanda.io/reference#integrating-monitoring-systems).
+    - **Insecure Skip Verify**: Check to skip TLS certificate chain and host
```
Comment (kelseiv) [inline content/chronograf/v1.9/guides/configuring-alert-endpoints.md:83]:
```suggestion
    - **Insecure Skip Verify**: Required if using a self-signed TLS certificate. Select to skip TLS certificate chain and host
```

File: content/chronograf/v1.9/guides/configuring-alert-endpoints.md [hunk 2]
```diff
@@ -60,101 +58,91 @@ For example, Chronograf's Slack integration allows users to specify a default ch
...
+      verification when connecting over HTTPS.
+      Required if using a self-signed TLS certificate.
```
Comment (kelseiv) [inline content/chronograf/v1.9/guides/configuring-alert-endpoints.md:85]:
```suggestion
```

File: content/chronograf/v1.9/guides/configuring-alert-endpoints.md [hunk 3]
```diff
@@ -60,101 +58,91 @@ For example, Chronograf's Slack integration allows users to specify a default ch
...
+    - **Configuration Enabled**: Check to enable configuration.
```
Comment (kelseiv) [inline content/chronograf/v1.9/guides/configuring-alert-endpoints.md:86]:
```suggestion
    - **Configuration Enabled**: Select to enable configuration.
```

File: content/chronograf/v1.9/guides/configuring-alert-endpoints.md [hunk 4]
```diff
@@ -60,101 +58,91 @@ For example, Chronograf's Slack integration allows users to specify a default ch
...
-  * **SSL CA**: Path to the SSL CA (certificate authority) file.
-  * **SSL Cert**: Path to the SSL host certificate.
-  * **SSL Key**: Path to the SSL certificate private key file.
-  * **Insecure Skip Verify**: Check to use SSL, but skip chain and host verification. Required if using a self-signed certificate.
+    - **ID**: Unique identifier for a Kafka cluster. Default is `localhost`.
+    - **Brokers**: List of Kafka broker addresses, using the `host:port` format.
+    - **Timeout**: Maximum amount of time to wait before flushing an incomplete batch. Default is `10s`.
+    - **Batch Size**: Number of messages batched before sending to Kafka. Default is `100`.
+    - **Batch Timeout**: Timeout period for the batch. Default is `1s`.
+    - **Use SSL**: Check to enable SSL communication.
```
Comment (kelseiv) [inline content/chronograf/v1.9/guides/configuring-alert-endpoints.md:104]:
```suggestion
    - **Use SSL**: Select to enable SSL communication.
```

File: content/chronograf/v1.9/guides/configuring-alert-endpoints.md [hunk 5]
```diff
@@ -60,101 +58,91 @@ For example, Chronograf's Slack integration allows users to specify a default ch
...
+    - **SSL CA**: Path to the SSL CA (certificate authority) file.
+    - **SSL Cert**: Path to the SSL host certificate.
+    - **SSL Key**: Path to the SSL certificate private key file.
+    - **Insecure Skip Verify**: Check to skip TLS certificate chain and host
```
Comment (kelseiv) [inline content/chronograf/v1.9/guides/configuring-alert-endpoints.md:108]:
```suggestion
    - **Insecure Skip Verify**: Required if using a self-signed TLS certificate. Select to skip TLS certificate chain and host
```

File: content/chronograf/v1.9/guides/configuring-alert-endpoints.md [hunk 6]
```diff
@@ -60,101 +58,91 @@ For example, Chronograf's Slack integration allows users to specify a default ch
...
+      verification when connecting over HTTPS.
+      Required if using a self-signed TLS certificate.
```
Comment (kelseiv) [inline content/chronograf/v1.9/guides/configuring-alert-endpoints.md:110]:
```suggestion
```

File: content/chronograf/v1.9/guides/configuring-alert-endpoints.md [hunk 7]
```diff
@@ -60,101 +58,91 @@ For example, Chronograf's Slack integration allows users to specify a default ch
...
+    - **Configuration Enabled**: Check to enable configuration.
 
 3. Click **Save Changes** to save the configuration settings.
 4. Click **Send Test Alert** to verify the configuration.
 
-See [Kafka event handler (Kapacitor)](/{{< latest "kapacitor" >}}/event_handlers/kafka/) in the Kapacitor documentation for details about enabling OpsGenie services using TICKscripts.
+See [Kafka event handler (Kapacitor)](/{{< latest "kapacitor" >}}/event_handlers/kafka/)
+in the Kapacitor documentation for details about enabling Kafka services using TICKscripts.
```
Comment (kelseiv) [inline content/chronograf/v1.9/guides/configuring-alert-endpoints.md:117]:
```suggestion
To enable Kafka services using TICKscript, see [Kafka event handler (Kapacitor)](/{{< latest "kapacitor" >}}/event_handlers/kafka/).
```

File: content/chronograf/v1.9/guides/configuring-alert-endpoints.md [hunk 8]
```diff
@@ -60,101 +58,91 @@ For example, Chronograf's Slack integration allows users to specify a default ch
...
 
-  * **API Key**: API key (or GenieKey). The API Key can be found by signing into your [OpsGenie account](https://app.opsgenie.com/auth/login) and selecting the **Settings** menu option in the **Admin** menu.
-  * **Teams**: List of [OpsGenie teams](https://docs.opsgenie.com/docs/teams) to be alerted.
-  * **Recipients** field, enter the list of [OpsGenie team members](https://docs.opsgenie.com/docs/teams#section-team-members)) to receive alerts.
-  * **Select recovery action**. Specify one of the following actions to take when an alert recovers:
-
-     - Add a note to the alert
-     - Close the alert
+    - **API Key**: API key (or GenieKey).
+      The API Key can be found by signing into your [OpsGenie account](https://app.opsgenie.com/auth/login)
```
Comment (kelseiv) [inline content/chronograf/v1.9/guides/configuring-alert-endpoints.md:138]:
```suggestion
      To find the API key, sign into your [OpsGenie account](https://app.opsgenie.com/auth/login)
```

File: content/chronograf/v1.9/guides/configuring-alert-endpoints.md [hunk 9]
```diff
@@ -60,101 +58,91 @@ For example, Chronograf's Slack integration allows users to specify a default ch
...
+      and selecting the **Settings** menu option in the **Admin** menu.
```
Comment (kelseiv) [inline content/chronograf/v1.9/guides/configuring-alert-endpoints.md:139]:
```suggestion
      and select the **Settings** menu option in the **Admin** menu.
```

File: content/chronograf/v1.9/guides/configuring-alert-endpoints.md [hunk 10]
```diff
@@ -60,101 +58,91 @@ For example, Chronograf's Slack integration allows users to specify a default ch
...
+    - **Teams**: List of [OpsGenie teams](https://docs.opsgenie.com/docs/teams) to be alerted.
+    - **Recipients** List of [OpsGenie team members](https://docs.opsgenie.com/docs/teams#section-team-members)) to receive alerts.
+    - **Select recovery action**: Actions to take when an alert recovers:
+        - Add a note to the alert
+        - Close the alert
+    - **Configuration Enabled**: Check to enable configuration.
```
Comment (kelseiv) [inline content/chronograf/v1.9/guides/configuring-alert-endpoints.md:145]:
```suggestion
    - **Configuration Enabled**: Select to enable configuration.
```

File: content/chronograf/v1.9/guides/configuring-alert-endpoints.md [hunk 11]
```diff
@@ -167,224 +155,221 @@ See the [AlertNode (Kapacitor TICKscript node) - OpsGenie v2](/{{< latest "kapac
...
 
-  * **Routing Key**: GUID of your PagerDuty Events API V2 integration, listed as "Integration Key" on the Events API V2 integration's detail page. See [Create a new service](https://support.pagerduty.com/docs/services-and-integrations#section-create-a-new-service) in the PagerDuty documentation details on getting an "Integration Key" (`routing_key`).
-  * **PagerDuty URL**: URL used to POST a JSON body representing the event. This value should not be changed. Valid value is `https://events.pagerduty.com/v2/enqueue`.
-  * **Configuration Enabled**: Check to enable this configuration.
+    - **Routing Key**: GUID of your PagerDuty Events API V2 integration, listed as "Integration Key" on the Events API V2 integration's detail page. See [Create a new service](https://support.pagerduty.com/docs/services-and-integrations#section-create-a-new-service) in the PagerDuty documentation details on getting an "Integration Key" (`routing_key`).
+    - **PagerDuty URL**: URL used to POST a JSON body representing the event. This value should not be changed. Valid value is `https://events.pagerduty.com/v2/enqueue`.
+    - **Configuration Enabled**: Check to enable this configuration.
```
Comment (kelseiv) [inline content/chronograf/v1.9/guides/configuring-alert-endpoints.md:173]:
```suggestion
    - **Configuration Enabled**: Select to enable this configuration.
```

File: content/chronograf/v1.9/guides/configuring-alert-endpoints.md [hunk 12]
```diff
@@ -167,224 +155,221 @@ See the [AlertNode (Kapacitor TICKscript node) - OpsGenie v2](/{{< latest "kapac
...
 3. Click **Save Changes** to save the configuration settings.
 4. Click **Send Test Alert** to verify the configuration.
 
-See the [PagerDuty Events API V2 Overview](https://v2.developer.pagerduty.com/docs/events-api-v2) for details on the PagerDuty Events API and recognized event types (`trigger`, `acknowledge`, and `resolve`).
+See the [PagerDuty Events API V2 Overview](https://v2.developer.pagerduty.com/docs/events-api-v2)
+for details on the PagerDuty Events API and recognized event types (`trigger`, `acknowledge`, and `resolve`).
 
-See [AlertNode (Kapacitor TICKscript node) - PagerDuty v2](/{{< latest "kapacitor" >}}/nodes/alert_node/#pagerduty-v2) in the Kapacitor documentation for details about enabling a new "Generic API" service using TICKscripts.
+See [AlertNode (Kapacitor TICKscript node) - PagerDuty v2](/{{< latest "kapacitor" >}}/nodes/alert_node/#pagerduty-v2)
+in the Kapacitor documentation for details about enabling a new "Generic API" service using TICKscripts.
```
Comment (kelseiv) [inline content/chronograf/v1.9/guides/configuring-alert-endpoints.md:182]:
```suggestion
To enable a new "Generic API" service using TICKscript, see [AlertNode (Kapacitor TICKscript node) - PagerDuty v2](/{{< latest "kapacitor" >}}/nodes/alert_node/#pagerduty-v2).
```

File: content/chronograf/v1.9/guides/configuring-alert-endpoints.md [hunk 13]
```diff
@@ -167,224 +155,221 @@ See the [AlertNode (Kapacitor TICKscript node) - OpsGenie v2](/{{< latest "kapac
...
+**To configure a Sensu alert endpoint:**
 
-The URL for the Pushover API. The default value is `https://api.pushover.net/1/messages.json`.
+1. In the **Configure Alert Endpoints** of the **Configure Kapacitor Connection** page,
+   click the **Sensu** tab.
+2. Enter the following:
 
-#### Configuration Enabled
+    - **Source**: Event source. Default is `Kapacitor`.
+    - **Address**: Base URL of [Sensu API](https://docs.sensu.io/sensu-go/latest/migrate/#architecture).
```
Comment (kelseiv) [inline content/chronograf/v1.9/guides/configuring-alert-endpoints.md:210]: To be sure, is the base URL what Sensu refers to as the "sensu-backend" or "etcd datastore"?

File: content/chronograf/v1.9/guides/configuring-alert-endpoints.md [hunk 14]
```diff
@@ -167,224 +155,221 @@ See the [AlertNode (Kapacitor TICKscript node) - OpsGenie v2](/{{< latest "kapac
...
+    - **Configuration Enabled**: Check to enable configuration.
```
Comment (kelseiv) [inline content/chronograf/v1.9/guides/configuring-alert-endpoints.md:211]:
```suggestion
    - **Configuration Enabled**: Select to enable configuration.
```

File: content/chronograf/v1.9/guides/configuring-alert-endpoints.md [hunk 15]
```diff
@@ -167,224 +155,221 @@ See the [AlertNode (Kapacitor TICKscript node) - OpsGenie v2](/{{< latest "kapac
...
+1. In the **Configure Alert Endpoints** of the **Configure Kapacitor Connection** page,
+   click the **ServiceNow** tab.
+2. Enter the following:
 
-Click **Send Test Alert** to test your alert endpoint configuration.
+    - **URL**: ServiceNow API URL. Default is `https://instance.service-now.com/api/global/em/jsonv2`.
+    - **Source**: Event source.
+    - **Username**: ServiceNow username.
+    - **Password**: ServiceNow password.
+    - **Configuration Enabled**: Check to enable configuration.
```
Comment (kelseiv) [inline content/chronograf/v1.9/guides/configuring-alert-endpoints.md:228]:
```suggestion
    - **Configuration Enabled**: Select to enable configuration.
```

File: content/chronograf/v1.9/guides/configuring-alert-endpoints.md [hunk 16]
```diff
@@ -167,224 +155,221 @@ See the [AlertNode (Kapacitor TICKscript node) - OpsGenie v2](/{{< latest "kapac
...
+      or configured in the **Slack Webhook**.
+    - **Configuration Enabled**: Check to enable configuration.
 
-Your bot's username must end in `bot`.
-For example: `mushroomKap_bot`.
+3. Click **Save Changes** to save the configuration settings.
+4. Click **Send Test Alert** to verify the configuration.
 
-`BotFather` responds:
+#### Add another Slack configuration
```
Comment (kelseiv) [inline content/chronograf/v1.9/guides/configuring-alert-endpoints.md:255]:
```suggestion
```

File: content/chronograf/v1.9/guides/configuring-alert-endpoints.md [hunk 17]
```diff
@@ -167,224 +155,221 @@ See the [AlertNode (Kapacitor TICKscript node) - OpsGenie v2](/{{< latest "kapac
...
+2. Enter the following:
 
-Your newly-created bot will appear in the chat list on the left side of the application.
+    - **SMTP Host**: SMTP host. Default is `localhost`.
+    - **SMTP Port**: SMTP port. Default is `25`.
+    - **From Email**: Email address to send messages from.
+    - **To Email**: Email address to send messages to.
+    - **User**: SMTP username.
+    - **Password**: SMTP password.
+    - **Configuration Enabled**: Check to enable configuration.
```
Comment (kelseiv) [inline content/chronograf/v1.9/guides/configuring-alert-endpoints.md:278]:
```suggestion
    - **Configuration Enabled**: Select to enable configuration.
```

File: content/chronograf/v1.9/guides/configuring-alert-endpoints.md [hunk 18]
```diff
@@ -167,224 +155,221 @@ See the [AlertNode (Kapacitor TICKscript node) - OpsGenie v2](/{{< latest "kapac
...
-If you can't find the API access token, create a new token with the steps
-below:
+1. In the **Configure Alert Endpoints** of the **Configure Kapacitor Connection** page,
+   click the **Talk** tab.
+2. Enter the following:
 
-1. Send `/token` to `@BotFather`
+    - **URL**: Talk API URL.
+    - **Author Name**: Message author name.
+    - **Configuration Enabled**: Check to enable configuration.
```
Comment (kelseiv) [inline content/chronograf/v1.9/guides/configuring-alert-endpoints.md:293]:
```suggestion
    - **Configuration Enabled**: Select to enable configuration.
```

File: content/chronograf/v1.9/guides/configuring-alert-endpoints.md [hunk 19]
```diff
@@ -167,224 +155,221 @@ See the [AlertNode (Kapacitor TICKscript node) - OpsGenie v2](/{{< latest "kapac
...
+**To configure a Microsoft Teams alert endpoint:**
 
-    For a description of the Bot API, see this page: https://core.telegram.org/bots/api
+1. In the **Configure Alert Endpoints** of the **Configure Kapacitor Connection** page,
+   click the **Teams** tab.
+2. Enter the following:
 
-#### Chat ID
+    - **Channel URL**: Microsoft Teams channel URL.
+    - **Configuration Enabled**: Check to enable configuration.
```
Comment (kelseiv) [inline content/chronograf/v1.9/guides/configuring-alert-endpoints.md:307]:
```suggestion
    - **Configuration Enabled**: Select to enable configuration.
```

File: content/chronograf/v1.9/guides/configuring-alert-endpoints.md [hunk 20]
```diff
@@ -167,224 +155,221 @@ See the [AlertNode (Kapacitor TICKscript node) - OpsGenie v2](/{{< latest "kapac
...
 
-Send a message to your bot in the Telegram application.
-The message text can be anything; your chat history must include at least one message to get your chat ID.
+    - **Token**:
+    - **Chat ID**: 
+    - **Select the alert message format**: Telegram message format
+        - Markdown _(default)_
+        - HTML
+    - **Disable link previews**: Disable [link previews](https://telegram.org/blog/link-preview) in Telegram messages.
+    - **Disable notifications**: Disable notifications on iOS Devices and sounds on Android devices.
```
Comment (kelseiv) [inline content/chronograf/v1.9/guides/configuring-alert-endpoints.md:327]:
```suggestion
    - **Disable notifications**: Disable notifications on iOS devices and sounds on Android devices.
```

File: content/chronograf/v1.9/guides/configuring-alert-endpoints.md [hunk 21]
```diff
@@ -167,224 +155,221 @@ See the [AlertNode (Kapacitor TICKscript node) - OpsGenie v2](/{{< latest "kapac
...
+      Android users will continue to receive notifications.
+    - **Configuration Enabled**: Check to enable configuration.
```
Comment (kelseiv) [inline content/chronograf/v1.9/guides/configuring-alert-endpoints.md:329]:
```suggestion
    - **Configuration Enabled**: Select to enable configuration.
```

File: content/chronograf/v1.9/guides/configuring-alert-endpoints.md [hunk 22]
```diff
@@ -167,224 +155,221 @@ See the [AlertNode (Kapacitor TICKscript node) - OpsGenie v2](/{{< latest "kapac
...
 
-```
-    {"ok":true,"result":[{"update_id":XXXXXXXXX,
-    "message":{"message_id":2,"from":{"id":123456789,"first_name":"Mushroom","last_name":"Kap"},"chat":{"id":123456789,"first_name":"Mushroom","last_name":"Kap","type":"private"},"date":1487183963,"text":"hi"}}]}
-```
+    - **API Key**: VictorOps API key.
+    - **Routing Key**: VictorOps [routing key](https://help.victorops.com/knowledge-base/routing-keys/).
+    - **VictorOps URL**: VictorOps alert API URL.
+      Default is `https://alert.victorops.com/integrations/generic/20131114/alert`.
+    - **Configuration Enabled**: Check to enable configuration.
```
Comment (kelseiv) [inline content/chronograf/v1.9/guides/configuring-alert-endpoints.md:343]:
```suggestion
    - **Configuration Enabled**: Select to enable configuration.
```

File: content/chronograf/v1.9/guides/configuring-alert-endpoints.md [hunk 23]
```diff
@@ -167,224 +155,221 @@ See the [AlertNode (Kapacitor TICKscript node) - OpsGenie v2](/{{< latest "kapac
...
+      Default is `add_event`.
+    - **Event Type**: Event type. Default is `rpc`.
+    - **Event TID**: Temporary request transaction ID. Default is `1`.
+    - **Collector Name**: Zenoss collector name. Default is `Kapacitor`.
+    - **Kapacitor to Zenoss Severity Mapping**: Map Kapacitor severities to [Zenoss severities](https://help.zenoss.com/docs/using-collection-zones/event-management/event-severity-levels).
+        - **OK**: Clear _(default)_
+        - **Info**: Info _(default)_
+        - **Warning**: Warning _(default)_
+        - **Critical**: Critical _(default)_
+    - **Configuration Enabled**: Check to enable configuration.
```
Comment (kelseiv) [inline content/chronograf/v1.9/guides/configuring-alert-endpoints.md:372]:
```suggestion
    - **Configuration Enabled**: Select to enable configuration.
```

**Review by kelseiv — APPROVED** [review]

File: content/chronograf/v1.9/guides/configuring-alert-endpoints.md (hunk 13 above)
Comment (sanderson) [inline content/chronograf/v1.9/guides/configuring-alert-endpoints.md:210]:
```suggestion
    - **Address**: URL of [Sensu HTTP API](https://docs.sensu.io/sensu-go/latest/migrate/#architecture).
```

**Pull request merged**